api-admin/src/main/java/bio/terra/pearl/api/admin/service/forms/SurveyExtService.java

```diff
@@ -41,6 +41,17 @@ public Survey get(String portalShortcode, String stableId, int version, AdminUse
     return survey;
   }
 
+  public List<Survey> listVersions(String portalShortcode, String stableId, AdminUser adminUser) {
+    Portal portal = authUtilService.authUserToPortal(adminUser, portalShortcode);
+    authSurveyToPortal(portal, stableId, 1); // TODO: this assumes there's always a v1. Bad?
```

Comment:
I can't imagine that there's any reason to check that all survey versions are accessible, so checking the first version seems like it should be enough. But I'm not sure if it's dangerous to assume that there's always a v1.

Perhaps a better route is to slightly change the behavior of `authSurveyToPortal`, so it doesn't require a specific version number.

Reply:
Interesting problem. `authSurveyToPortal` just loads the survey to make sure it exists, verifying that the portal/stableId/version combination is valid. You _could_ add a variant of `authSurveyToPortal` that checks if there are _any_ survey versions, but that would eventually call `SurveyDao.findByStableId` which would fetch _all_ the data... and all this to `findByStableIdNoContent` below because you don't need all of the data.

Maybe we need a `boolean` `existsByStableId` to use for auth checks.

Reply:
Yeah, let's move the portal id check after the findByStableId so we can check all the returned surveys -- it'll be more correct and more performant.  Suggestion below